openssl/src/x509/mod.rs

```diff
@@ -1827,6 +1849,157 @@ impl X509Crl {
         X509Crl,
         ffi::d2i_X509_CRL
     }
+
+    pub fn new(issuer_cert: &X509) -> Result<Self, ErrorStack> {
+        unsafe {
+            let crl = Self(cvt_p(ffi::X509_CRL_new())?);
+            #[cfg(ossl110)]
+            {
+                cvt(ffi::X509_CRL_set_version(
+                    crl.as_ptr(),
+                    issuer_cert.version() as c_long,
```

Comment:
This makes no sense. The issuer cert will most likely have version 3 because of extensions being present, but there is no version 3 for CRLs. I would just set it to 2 (i.e., pass 1) since you allow setting the crlNumber extension (in which case the version must be 2).

Also, why is this not enabled for libressl? X509_CRL_set_version is available since LibreSSL 2.5.1.

Reply:
Thank you very much for the feedback.

The version was just a silly mistake, I now check the issuer_cert's version and build a corresponding CRL. I am not too sure if supporting old versions is a good idea, but openssl does it.. (I am open to suggestions on this; I think the alternative would be to reject old certs/return an error).

I ran into some version troubles when fixing the checks. get_version used to be a macro, apparently. I copied X509's get_version approach and simply do not provide the functionality. Sadly most of the changes are related to this.
Since we intend to use a recent version of openssl, it is not a problem for us, but I'd love to extend availability if possible with reasonable effort.